x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/types.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { UserConfiguredActionConnector } from '../../../../types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { ExecutorSubActionPushParams } from '../../../../../../actions/server/builtin_action_types/swimlane/types';
+
+export type SwimlaneActionConnector = UserConfiguredActionConnector<
+  SwimlaneConfig,
+  SwimlaneSecrets
+>;
+
+export interface SwimlaneConfig {
+  apiUrl: string;
+  appId: string;
+  connectorType: SwimlaneConnectorType;
+  mappings: SwimlaneMappingConfig;
+}
+
+export interface SwimlaneMappingConfig {
+  alertIdConfig: SwimlaneFieldMappingConfig;
+  severityConfig: SwimlaneFieldMappingConfig;
+  caseNameConfig: SwimlaneFieldMappingConfig;
+  caseIdConfig: SwimlaneFieldMappingConfig;
+  ruleNameConfig: SwimlaneFieldMappingConfig;
+  commentsConfig: SwimlaneFieldMappingConfig;
+  descriptionConfig: SwimlaneFieldMappingConfig;
+  [key: string]: SwimlaneFieldMappingConfig;
```

Comment:
I'd guess that for DX, it would be better to move the `[key: string]` part of this definition under a new top-level property as a peer of the other `*Config` keys.  Otherwise we'll miss out on some type checking, in case someone misspells the "fixed" names.  For example, if someone wants to reference `alertIdConfig`, but use `alertLdConfig`, that's not a type error, but you'd probably want it flagged that way.

Not completely clear to me how these fields would be used though, seems like it would require a change to the looping I've seen through these config objects.  So may not be worth it.

Reply:
You are right! I tried to improved it. Let me know if you think of a better way.